### PR TITLE
Api communication

### PR DESCRIPTION
**Pull request by domTombola**
got api communication working and can play the noughts and crosses game.

### CODE REVIEW

File: main-app/app/htmlTemplates/gameboard.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<table id="gameboard" >
+    <tr>
+        <td class="player{{gameboard[0]}}" ng-click="gameboardTapped(0)"><img src="images/blank.png"></td>
```
Comment (keithbarrow) [inline main-app/app/htmlTemplates/gameboard.html]: food for thought - this is repetitive, and a construct we haven't covered yet will simplify this code down for you...

File: main-app/app/less/main.less [hunk 2]
```diff
@@ -12,4 +12,71 @@ a{
...
+.cellBorder {
+  border-left: 1px solid @link-color;
+  border-right: 1px solid @link-color;
+}
+
+
+
+//Styles for the image sprites
+//.player1, .player2 {
+//  background: url('../images/noughtsAndCrosses.png') no-repeat;
```
Comment (keithbarrow) [inline main-app/app/less/main.less]: Comments again - the section headers in less files are a good idea, the organise things a bit and are removed at compile. 
But the commented source is a problem - in a real project someone will need to maintain & work out why you've left them here.

File: main-app/app/scripts/Api Service/GameApi.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        };
+
+        this.makeGameMove = function (currentPlayer, gridIndex) {
+            var data = {
+                'playerNumber': currentPlayer,
+                'chosenSquare': gridIndex
+            };
+            return callApi('makemove', data);
+        };
+    }]);
```
Comment (keithbarrow) [inline main-app/app/scripts/Api Service/GameApi.js]: Good.

File: main-app/app/scripts/directives/gameState.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.NoughtAndCrosses')
```
Comment (keithbarrow) [inline main-app/app/scripts/directives/gameState.js]: Why directives? Both this and (more so) the game board represent decent chunks of functionality. There is a better construct out there.

File: main-app/app/scripts/Controllers/GameboardController.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+(function () {
```
Comment (keithbarrow) [inline main-app/app/scripts/Controllers/GameboardController.js]: This needs breaking down - assuming first cut for now.

File: main-app/app/less/main.less (hunk 2 above)
Comment (domTombola) [inline main-app/app/less/main.less]: sorted this out in next pull request

File: main-app/app/scripts/directives/gameState.js (hunk 4 above)
Comment (domTombola) [inline main-app/app/scripts/directives/gameState.js]: removed those two directives and using views instead in next pull request

**Pull request merged**